coral/src/app/features/topics/schema-request/TopicSchemaRequest.tsx

```diff
@@ -78,6 +84,20 @@ function TopicSchemaRequest(props: TopicSchemaRequestProps) {
   >({
     queryKey: ["schemaRegistryEnvironments"],
     queryFn: () => getEnvironmentsForSchemaRequest(),
+    onSuccess: (environments) => {
+      const currentEnv = searchParams.get("env");
+      if (currentEnv) {
+        const isValidEnv =
+          environments.find((env) => currentEnv === env.id) !== undefined;
+        if (!isValidEnv) {
+          navigate(-1);
+        } else {
+          console.log("currentEnv", currentEnv);
+          setPresetEnvironment(currentEnv);
```

Comment:
Yes, you're right. I added the state to make sure we're setting the select element to readonly if there is a _valid_ env, but that's unnecessary as I realise now, because we redirect if it's not valid! Will update, thank you!